pkgs/build-support/setup-hooks/patch-shebangs.sh

```diff
@@ -50,29 +50,29 @@ patchShebangs() {
     while IFS= read -r -d $'\0' f; do
         isScript "$f" || continue
 
-        oldInterpreterLine=$(head -1 "$f" | tail -c+3)
-        read -r oldPath arg0 args <<< "$oldInterpreterLine"
+        read -r oldInterpreterLine < "$f"
+        read -r oldPath arg0 args <<< "${oldInterpreterLine:3}"
 
-        if [ -z "$pathName" ]; then
-            if [ -n "$strictDeps" ] && [[ "$f" = "$NIX_STORE"* ]]; then
+        if [[ -z "$pathName" ]]; then
+            if [[ -n "$strictDeps" ]] && [[ "$f" = "$NIX_STORE"* ]]; then
                 pathName=HOST_PATH
             else
                 pathName=PATH
             fi
         fi
 
-        if $(echo "$oldPath" | grep -q "/bin/env$"); then
+        if [[ "$oldPath" == *"/bin/env" ]]; then
             # Check for unsupported 'env' functionality:
             # - options: something starting with a '-'
             # - environment variables: foo=bar
-            if $(echo "$arg0" | grep -q -- "^-.*\|.*=.*"); then
+            if [[ "$arg0" == "-"* ]] || [[ "$arg0" == *"="* ]]; then
```

Comment:
Could simplify to
```suggestion
            if [[ $arg0 == "-"* || $arg0 == *"="* ]]; then
```
or even (I believe)
```suggestion
            if [[ $arg0 == -* || $arg0 == *=* ]]; then
```
but I think the first is a bit more clear.

Reply:
I did not even no this was possible.